GOOD-I-DEER/nodes/face-vectorization-node/face-vectorization.js

```diff
@@ -15,72 +15,99 @@ module.exports = function (RED) {
     node.on("input", async function (msg) {
       const inputData = msg.payload;
 
-      const vectors = [];
-      if (config.inputType == 0) {
-        vectors.push(await image_vectorization(inputData));
-      } else if (config.inputType == 1) {
-        for (let i = 0; i < inputData.length; ++i) {
-          vectors.push(await image_vectorization(inputData[i]));
+      try {
+        const vectors = [];
+        if (config.inputType == 0) {
+          vectors.push(await image_vectorization(inputData));
+        } else if (config.inputType == 1) {
+          for (let i = 0; i < inputData.length; ++i) {
+            vectors.push(await image_vectorization(inputData[i]));
+          }
         }
-      }
-
-      if (config.returnType == 0) {
-        msg.payload = vectors;
-        node.send(msg);
-      } else if (config.returnType == 1) {
-        const filePath = config.path;
-        const textData = JSON.stringify(vectors);
 
-        const dir = path.dirname(filePath);
-        if (!fs.existsSync(dir)) {
-          fs.mkdirSync(dir, { recursive: true });
+        if (config.returnType == 0) {
+          msg.payload = vectors;
+          node.send(msg);
+        } else if (config.returnType == 1) {
+          const filePath = config.path;
+          const textData = JSON.stringify(vectors);
+
+          const dir = path.dirname(filePath);
+
+          if (!fs.existsSync(dir)) {
+            fs.mkdirSync(dir, { recursive: true });
+          }
+
+          if (config.method == 0) {
+            fs.writeFileSync(filePath, textData, "utf-8");
+          } else if (config.method == 1) {
+            let existData = fs.readFileSync(filePath, "utf-8");
+            const allData = JSON.parse(existData);
+            allData.push(...vectors);
+            fs.writeFileSync(filePath, JSON.stringify(allData), "utf-8");
+          }
         }
-
-        fs.writeFileSync(filePath, textData, "utf-8");
+      } catch (error) {
+        node.error("An error occurred: " + error.message.split("\n")[0]);
       }
     });
 
     async function image_vectorization(inputData) {
-      const model = await onnx.InferenceSession.create(modelPath);
+      try {
+        const model = await onnx.InferenceSession.create(modelPath);
 
-      const inputName = model.inputNames[0];
+        const inputName = model.inputNames[0];
 
-      const inputDims = [1, 3, 224, 224];
+        const inputDims = [1, 3, 224, 224];
```

Comment:
Loading the model and defining settings only needs to be done once! 😘

Reply:
This is evidence of an attempt to allow users to choose from a variety of models. I plan to leave it for expandability.